File: Source/UnitTests/Translator/IntegrationTest/IKVMExpected/TestClass.cs
namespace Test.Integration
{
	using java.lang;

	using TestCase = junit.framework.TestCase;

	[NUnit.Framework.TestFixture()]
	public class TestClass : TestCase
	{

		protected override void setUp()
		{
		}

		protected override void tearDown()
		{
		}

		[NUnit.Framework.Test()]
		public void TestMethod()
		{
			AbstractClass abc = null;
			string testName = abc.Name;
			abc.Name = testName;
			Integer a = new Integer(0);
			int i = a.intValue();
			AbstractClass.InnerAbstractClass inf = new AnonymousClassAbstractClass_InnerAbstractClass1(i, this);
			Integer[] b = null;
			bool ab = (a == null) || (b == null);
			int j = 0;
			assertEquals("message :", i, j);
		}
		private class AnonymousClassAbstractClass_InnerAbstractClass1 : AbstractClass.InnerAbstractClass
		{
			public AnonymousClassAbstractClass_InnerAbstractClass1(int num, TestClass enclosingInstance) : base(num)
			{
				this.enclosingInstance = enclosingInstance;
			}
			public override void InnerAbstractClassMethod(int num)
			{
			}
			private TestClass enclosingInstance;
			public TestClass Enclosing_Instance {
				get { return enclosingInstance; }
			}
		}

	}
}

File: Source/UnitTests/Translator/IntegrationTest/Base/TestClass.java
package Test.Integration;

import junit.framework.TestCase;

public class TestClass extends TestCase
{
    public TestClass(int arg1, String arg2)
    {
    }

    public void setUp()
    {
    }

    public void tearDown()
    {
    }

    public void testMethod()
    {
		AbstractClass abc = null;
		String testName = abc.getName();
		abc.setName(testName);
		Integer a = new Integer(0);
		int i = a.intValue();
        AbstractClass.InnerAbstractClass inf = new AbstractClass.InnerAbstractClass(i)
                                                    {
                                                        public void InnerAbstractClassMethod(int num)
                                                        {
                                                        }
                                                    };
        Integer b[] = null;
        boolean ab = (a == null) || (b == null);
		int j = 0;
        assertEquals("message :", i, j);
    }

    public void main()
    {
    }
}
